samcli/lib/utils/resource_trigger.py

```diff
@@ -103,22 +103,28 @@ def get_dir_path_handler(dir_path: Path, ignore_regexes: Optional[List[str]] = N
 
 class TemplateTrigger(ResourceTrigger):
     _template_file: str
+    _stack_name: str
     _on_template_change: OnChangeCallback
     _validator: DefinitionValidator
 
-    def __init__(self, template_file: str, on_template_change: OnChangeCallback) -> None:
+    def __init__(self, template_file: str, stack_name: str, on_template_change: OnChangeCallback) -> None:
         """
         Parameters
         ----------
         template_file : str
             Template file to be watched
+        stack_name: str
+            Stack name of the template
         on_template_change : OnChangeCallback
             Callback when template changes
         """
         super().__init__()
         self._template_file = template_file
+        self._stack_name = stack_name
         self._on_template_change = on_template_change
         self._validator = DefinitionValidator(Path(self._template_file))
+        if not self._validator.raw_validate():
+            raise InvalidTemplateFile(self._template_file, self._stack_name)
```

Comment:
This will break a use case where user will have an invalid template and start `sam sync --watch` and that template file won't be watched. If user fixes their template and expects sam to sync it, it won't happen. So I think we shouldn't be failing for invalid templates here.